### PR TITLE
test: refactor rpc_signrawtransaction and add logging

### PR DESCRIPTION
**Pull request by jonatack**
As a follow-up to #18484, the new tests are good but bury the one non-duplicate line in each test that sets the witness script, and there is no logging in the testfile. This PR makes it easy to see what is unique to each of the new tests and adds logging.

### CODE REVIEW

File: test/functional/rpc_signrawtransaction.py [hunk 1]
```diff
@@ -169,29 +169,18 @@ def witness_script_test(self):
...
-        self.nodes[0].generate(1)
-        # Now create and sign a transaction spending that output on node[0], which doesn't know the scripts or keys
-        spending_tx = self.nodes[0].createrawtransaction([{'txid': txid, 'vout': vout}], {self.nodes[1].getnewaddress(): Decimal("9.999")})
-        spending_tx_signed = self.nodes[0].signrawtransactionwithkey(spending_tx, [embedded_privkey], [{'txid': txid, 'vout': vout, 'scriptPubKey': script_pub_key, 'redeemScript': redeem_script, 'witnessScript': witness_script, 'amount': 10}])
-        # Check the signing completed successfully
-        assert 'complete' in spending_tx_signed
-        assert_equal(spending_tx_signed['complete'], True)
-        self.nodes[0].sendrawtransaction(spending_tx_signed['hex'])
+        # Now test with P2PKH and P2PK scripts as the witnessScript
+        for tx_type in ['P2PKH', 'P2PK']:  # these tests are order-independant
```
Comment (theStack) [inline test/functional/rpc_signrawtransaction.py:176]: nit: s/independant/independent

File: test/functional/rpc_signrawtransaction.py [hunk 2]
```diff
@@ -50,6 +51,7 @@ def successful_signing_test(self):
 
     def test_with_lock_outputs(self):
         """Test correct error reporting when trying to sign a locked output"""
```
Comment (theStack) [inline test/functional/rpc_signrawtransaction.py:53]: This comment is redudant to the introduced logging message and could be removed? (Same as for `witness_script_test()` below).

**Review by theStack — COMMENTED** [review]
ACK modulo nits
Deduplication and better logging are always good :+1: 

For other reviewers, I suggest looking at the refactor diff with a larger number of lines of context via the `-U` option (at a first glance it looks like the signing part has been removed, as one doesn't see the whole newly introduced function `verify_txn_with_witness_script()` to the end), e.g.
```git show -U20 269b664ced```

File: test/functional/rpc_signrawtransaction.py (hunk 1 above)
Comment (jonatack) [reply test/functional/rpc_signrawtransaction.py]: good eye -- done

File: test/functional/rpc_signrawtransaction.py (hunk 2 above)
Comment (jonatack) [reply test/functional/rpc_signrawtransaction.py]: done

**Comment by jonatack** [comment]
Thanks for reviewing, @theStack. I've updated with both of your suggestions.

**Review by theStack — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/pull/18545/commits/9cdddae3b4efee071d71ba3b6629a53017332f6f :egg: :rabbit:

**Pull request merged**